components/eamxx/src/diagnostics/tests/field_at_height_tests.cpp

```diff
@@ -98,7 +110,18 @@ TEST_CASE("field_at_height")
     const auto& dims = f.get_header().get_identifier().get_layout().dims();
     for (int i=0; i<dims[0]; ++i) {
       for (int j=0; j<dims[1]; ++j) {
-        v(i,j) = nlevs-j;
+        v(i,j) = dims[1]-j+i;
```

Comment:
Ah, now I understand why you mix j and i in `get_ref_field`. The surf elevation is basically `z_surf(i) = 1 + i`.

I would make the test clearer (or else it may be a bit of a headache for those who will maintain it down the road):
1. create the z_surf field.
2. create z_int as z_surf + ilev
3. create z_mid as z_surf + ilev + 0.5 (the half is to make it consistent with z_int, for clarity).
4. pass z (mid or int) to `get_ref_field`, and do something like
```
  if (z_lev < z_surf) {
    ...
```

Reply:
I can do that.  I think having a +0.5 for `z_mid` will break the "simplicity" of the test just taking a subfield of the data at the z_tgt level.  But I think that is okay, the test might be more robust if we actually triggered random locations (rather than integers) for the z_tgt and thus for the interpolation.

Reply:
yeah, that's fine. You can keep `z_mid = z_surf + ilev`. For the sake of this testing, we don't need to have mid to be in the middle of interfaces...